packages/react-data-grid/src/common/editors/TextEditor.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { forwardRef } from 'react';
+
+import { EditorProps } from '../types';
+
+type SimpleTextEditorProps = Pick<EditorProps<string>, 'value' | 'onChange' | 'onCommit'>;
+
+export default forwardRef<HTMLInputElement, SimpleTextEditorProps>(function TextEditor({
```

Comment:
I took a quick look at your structure. My main concern is that for the editors provided by others, they must use `forwardRef` or a class component, i.e. a ref is still needed for RDG API. For mine, it might be similar, the user must hook the `useDataGridEditor` to their editor. and if their editor was not a class component, they might have to use render props pattern to delegate the keyboard event to their component.

Reply:
I added an `inputRef` prop to make it explicit. This way it can be used with the class component as well. Take a look and let me know what you think. I will check your changes in the meantime.